conans/client/tools/git.py

```diff
@@ -0,0 +1,55 @@
+import subprocess
+from conans import tools
+from conans.errors import ConanException
+
+
+def git_uncommitted(folder):
+    with tools.chdir(folder):
+        try:
+            command = "git diff-index HEAD --quiet --"
+            subprocess.check_output(command, shell=True)
+            return False
+        except subprocess.CalledProcessError:
+            return True
+
+
+def git_origin(folder):
+    with tools.chdir(folder):
+        try:
+            command = "git remote -v"
+            remotes = subprocess.check_output(command, shell=True).decode().strip()
+            for remote in remotes.splitlines():
+                try:
+                    name, url = remote.split(None, 1)
+                    url, _ = url.rsplit(None, 1)
+                    if name == "origin":
+                        return url
+                except Exception:
+                    pass
+        except subprocess.CalledProcessError:
+            pass
+    return None
+
+
+def git_branch(folder):
+    with tools.chdir(folder):
+        command = "git branch"
+        try:
+            branches = subprocess.check_output(command, shell=True).decode().strip()
+            branches = [branch.replace("*", "").strip() for branch in branches.splitlines()
+                        if branch.lstrip()[0] == "*"]
+            branch = branches[0]
+            return branch
+        except Exception as e:
+            raise ConanException("Unable to get git branch from %s\n%s" % (folder, str(e)))
+
+
+def git_commit(folder):
```

Comment:
Naming is ambiguous. Expecting that this tool will be extended in the next versions (e.g. to support commiting  local changes). Better to rename to `get_rev()` and implement the whole tool as GIT class (reuse the folder argument, `git_exec()` method for language enforcement, origin url setup to handle credentials, etc.)

Reply:
Yes, agree. 
Also, I am naming the ``scm = { "checkout": "source | branch | commit | tag" } as something that basically git can perform the ``checkout``. Do you think it should be named ``revision``?

Reply:
I think `revision` is more obvious instead of `commit`. Probably `checkout` could also be renamed to something like `scm_behavior` or `scm_action` or `checkout_action` or something like that, because it describes what to do or how to handle the other scm information. On the other hand one can say `checkout branch` or `checkout tag`. But then you should rename `source` to `none` (to `checkout none`).
How is your current plan for the rest of the dictionary. Can you provide a complete example?